services/TipiVariable.py

```diff
@@ -29,25 +27,28 @@ def processRequest(self, message):
         ti_message = message.split( chr(0x1e) )
         
         caller_guid     = ti_message[0] if len(ti_message) >= 1 else ''    # Program's GUID
-        action          = ti_message[1] if len(ti_message) >= 2 else ''    # 'R', 'RS', 'W', 'U', 'T'   read, read-simple, write, transmit via UDP, transmit via TCP
-        queue           = ti_message[2] if len(ti_message) >= 3 else ''    # Allow values to queue for these variables. (only locally for now)
-        results_var     = ti_message[3] if len(ti_message) >= 4 else ''    # results_var (optional)
-        var_key1        = ti_message[4] if len(ti_message) >= 5 else ''    # var key
-        var_val1        = ti_message[5] if len(ti_message) >= 6 else ''    # var val
-        var_key2        = ti_message[6] if len(ti_message) >= 7 else ''    # var key
-        var_val2        = ti_message[7] if len(ti_message) >= 8 else ''    # var val
-        var_key3        = ti_message[8] if len(ti_message) >= 9 else ''    # var key
-        var_val3        = ti_message[9] if len(ti_message) >= 10 else ''   # var val
-        var_key4        = ti_message[10] if len(ti_message) >= 11 else ''  # var key
-        var_val4        = ti_message[11] if len(ti_message) >= 12 else ''  # var val
+        context         = ti_message[1] if len(ti_message) >= 2 else ''    # optional context, used when storing on myti99.com. Does Not Apply locally.
+        action          = ti_message[2] if len(ti_message) >= 3 else ''    # 'R', 'RS', 'W', 'U', 'T'   read, read-simple, write, transmit via UDP, transmit via TCP
+        queue           = ti_message[3] if len(ti_message) >= 4 else ''    # Allow values to queue for these variables. (only locally for now)
+        results_var     = ti_message[4] if len(ti_message) >= 5 else ''    # results_var (optional)
+        var_key1        = ti_message[5] if len(ti_message) >= 6 else ''    # var key
+        var_val1        = ti_message[6] if len(ti_message) >= 7 else ''    # var val
+        var_key2        = ti_message[7] if len(ti_message) >= 8 else ''    # var key
+        var_val2        = ti_message[8] if len(ti_message) >= 9 else ''    # var val
+        var_key3        = ti_message[9] if len(ti_message) >= 10 else ''    # var key
+        var_val3        = ti_message[10] if len(ti_message) >= 11 else ''  # var val
+        var_key4        = ti_message[11] if len(ti_message) >= 12 else ''  # var key
+        var_val4        = ti_message[12] if len(ti_message) >= 13 else ''  # var val
 
         response = str(results_var) if len(results_var) else ''
 
         # Load our existing variables into a dict:
         self.ti_vars = {}
+        self.ti_global = {}
         
         guid_file = runtime_dir + caller_guid
-        
+        global_file = runtime_dir + 'GLOBAL'
```

Comment:
Why do you need to have a 'GLOBAL' file, couldn't the guid used have just been 'GLOBAL' if that is what the application wants?

Reply:
The answer to this seems to be that SESSION_ID and REMOTE_HOST need to be read from GLOBAL even when the T action is happening for a guid of CHATTI